rust/src/ike/ike.rs

```diff
@@ -0,0 +1,628 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// Author: Frank Honza <frank.honza@dcso.de>
+
+extern crate ipsec_parser;
+use self::ipsec_parser::*;
+
+use std;
+use crate::core::{self, ALPROTO_UNKNOWN, ALPROTO_FAILED, AppProto, Flow, STREAM_TOSERVER, STREAM_TOCLIENT};
+use crate::log::*;
+use std::mem::transmute;
+use crate::applayer;
+use crate::applayer::*;
+use std::ffi::{CString, CStr};
+use nom;
+use crate::ike::parser::*;
+use crate::ike::ikev1::{handle_ikev1, IkeV1Header, Ikev1Container};
+use crate::ike::ikev2::{handle_ikev2, Ikev2Container};
+use std::collections::HashSet;
+
+static mut ALPROTO_IKE: AppProto = ALPROTO_UNKNOWN;
+
+#[repr(u32)]
+pub enum IkeEvent {
+    MalformedData = 0,
+    NoEncryption,
+    WeakCryptoEnc,
+    WeakCryptoPRF,
```

Comment:
Are these events part of the new PR ? Or were they already here ?

Reply:
no, were already there, i've just added PayloadExtraData.....